### PR TITLE
[Cluster.Hosting] Fix nullable warnings and modernize codes

### PR DESCRIPTION
**Pull request by Arkatufus**

### CODE REVIEW

File: src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs [hunk 1]
```diff
@@ -824,7 +824,7 @@ internal static Config CreateReceptionistConfig(string name, string? role)
                 .Append(root).Append("name:").AppendLine(name.ToHocon());
             
             if(!string.IsNullOrEmpty(role))
-                sb.Append(root).Append("role:").AppendLine(role.ToHocon());
+                sb.Append(root).Append("role:").AppendLine(role!.ToHocon());
```
Comment (Aaronontheweb) [inline src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs:827]: LGTM - `ToHocon` has `null` handling built into it, IIRC? @Arkatufus

**Review by Aaronontheweb — COMMENTED** [review]
Left a question

File: src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs (hunk 1 above)
Comment (Arkatufus) [reply src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs:827]: Its compiler/roslyn error, its already checked for null by the line before it.

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**